Invalid typings of `TableMetadata`

#### Environment details

  - OS: macOS 10.14.4
  - Node.js version: v11.12.0
  - npm version: v3.4.1
  - `@google-cloud/bigquery` version: 3.0.0

#### Steps to reproduce

Write this code:

```
import { TableSchema, TableMetadata } from "@google-cloud/bigquery/build/src/table";

let schema : TableSchema = { fields: [ { name: "price", type: "INTEGER" } ] };
let metadata: TableMetadata = { schema };
```

TypeScript Compiler says:
```
test.ts:4:33 - error TS2322: Type 'ITableSchema' is not assignable to type '(ITableSchema & string) | (ITableSchema & ITableFieldSchema[])'.
  Type 'ITableSchema' is not assignable to type 'ITableSchema & ITableFieldSchema[]'.
    Type 'ITableSchema' is missing the following properties from type 'ITableFieldSchema[]': length, pop, push, concat, and 26 more.

4 let metadata: TableMetadata = { schema };
                                  ~~~~~~

  node_modules/@google-cloud/bigquery/build/src/types.d.ts:1632:5
    1632     schema?: ITableSchema;
             ~~~~~~
    The expected type comes from property 'schema' which is declared here on type 'TableMetadata'


Found 1 error.
```

Inside the library, typings around `TableMetadata` is defined as:

```
type ITable = { schema?: ITableSchema, ... }
type TableSchema = ITableSchema = { fields?: ITableFieldSchema[] }
type TableMetadata = bigquery.ITable & { schema?: string | TableField[]; ...}
type TableField = ITableFieldSchema = { fields?: ITableFieldSchema[], name?: string, type?: string }
```

From definition above, TableMetadata can be inferred as:

```
TableMetadata = bigquery.ITable                               & { schema?: string | TableField[] }
              = { schema?: ITableSchema }                     & { schema?: string | TableField[] }
              = { schema?: { fields?: ITableFieldSchema[] } } & { schema?: TableField[] }
              = { schema?: { fields?: ITableFieldSchema[] } } & { schema?: ITableFieldSchema[] }
              = { schema?: { fields?: ITableFieldSchema[] } } & { schema?:  { fields?: ITableFieldSchema[], name?: string, type?: string }[] }
              = { schema?: { fields?: ITableFieldSchema[] } } & { schema?:  { fields?: ITableFieldSchema[] }[] }
```

There is a contradiction between `bigquery.ITable` and `{ schema?: string | TableField[]; }`, so that any value without `null` can't satisfy its type definition.

I have workaround by editing `TableMetadata` as `bigquery.ITable | { schema?: string | TableField[]; ... }`, not `bigquery.ITable & { schema?: string | TableField[]; ... }`, and it seems working.

--- a/src/table.ts
+++ b/src/table.ts
@@ -118,7 +118,7 @@ export type CopyTableMetadata = JobRequest<bigquery.IJobConfigurationTableCopy>;
 
 export type TableMetadata = bigquery.ITable & {
   name?: string;
-  schema?: string | TableField[];
+  schema?: string | TableField[] | TableSchema;
   partitioning?: string;
   view?: string | ViewDefinition;
 };

